src/containers/TrackList.js

```diff
@@ -1,9 +1,42 @@
 import { connect } from 'react-redux';
 import { getNewTracks, startSession } from '../actions';
-import TrackListView from '../components/TrackListView';
+import ViewableTrackList from '../components/ViewableTrackList';
+import _ from 'lodash';
+
+const filterByTrackName = (trackList, filters) => {
+  return filters.byTrackName === ''
+    ? trackList
+    : _.filter(trackList, o => {
+        return o.title
+          .toLowerCase()
+          .includes(filters.byTrackName.trimLeft().toLowerCase());
+      });
+};
+
+const filterByArtistName = (trackList, filters) => {
+  return filters.byArtistName === ''
+    ? trackList
+    : _.filter(trackList, o => {
+        return o.user.username
+          .toLowerCase()
+          .includes(filters.byArtistName.trimLeft().toLowerCase());
+      });
+};
```

Comment:
Future: DRY this up. Could you pass the filter name and use `_.partial`? You'll have to switch to [] syntax for pulling stuff out of the `filters` arg.